src/ssl.c

```diff
@@ -285,7 +287,16 @@ int ssl_check_cert(SSL *ssl, struct conf **confs, struct conf **cconfs)
 		NID_commonName, tmpbuf, sizeof(tmpbuf));
 	if(strcasecmp(tmpbuf, ssl_peer_cn))
 	{
-		logp("cert common name doesn't match configured ssl_peer_cn\n");
+		if(first)
+		{
+			logp("cert common name doesn't match configured ssl_peer_cn\n");
+			logp("'%s'!='%s'\n", tmpbuf, ssl_peer_cn);
+			logp("falling back to cname\n");
+			first=0;
+			ssl_peer_cn=get_string(cconfs[OPT_ORIG_CNAME]);
+			goto redo;
```

Comment:
My policy for 'goto' statements is to use them only to jump to the end of a function.
I think that it would be better to refactor this section a bit so that it is broken into smaller functions.
I think I will do this for you after I merge it though.